File: 2024/day06-1.js
// Puntuación: 4 estrellas

/** @param {string[]} gifts
 *  @returns {boolean} True if the gift is inside the box
 */
function inBox(box) {
  const gift = '*'

  // validamos que no está ni en la tapa ni en el fondo
  if( box[0].includes(gift) ) return false
  if( box.at(-1).includes(gift)) return false

  let inside = false

  for(const row of box) {
    // validamos que no está en los laterales
    if(row.startsWith(gift)) return false
    if(row.endsWith(gift)) return false

    // revisamos si está dentro o no
    if(row.includes(gift)) return true
  }

  return inside
}

File: 2024/day06-1.py
# Puntuación: 5 estrellas

def in_box(box):
  gift = '*';

  if(gift in box[0]) return False
  if(gift in box[-1]) return False

  rest = ''.join(str(x) for x in box[1,-1])
  found = gift in rest;
  if( !found ) return False

  width = box[0].length;
  mod = rest.index(gift) % width;
  return (mod > 0 && mod < width - 1)
